op-node/rollup/derive/channel_out.go

```diff
@@ -57,20 +64,35 @@ type ChannelOut struct {
 
 	// Compressor stage. Write input data to it
 	compress Compressor
-
+	// closed indicates if the channel is closed
 	closed bool
+	// batchType indicates whether this channel uses SingularBatch or SpanBatch
+	batchType uint
+	// spanBatchBuilder contains information requires to build SpanBatch
+	spanBatchBuilder *SpanBatchBuilder
```

Comment:
What's the API for building span batches - it kinda seems different enough that we should be using a `SpanChannelOut` rather than having two different sets of logic inside `ChannelOut`